arch/arm/core/cortex_m/mpu/nxp_mpu.c

```diff
@@ -209,165 +378,110 @@ void arm_core_mpu_disable(void)
 	SYSMPU->CESR |=  SYSMPU_CESR_SPERR_MASK;
```

Comment:
Add __DSB(); before /* Disable MPU */
Remove the barriers at the end of nxp_mpu_init(), this way barriers are always carried on when enabling/disabling mpu not only at mpu initialization.

Reply:
Sure.

Reply:
Added